src/main/java/com/minecolonies/coremod/colony/crafting/ItemStorageFactory.java

```diff
@@ -65,22 +80,28 @@ public ItemStorage deserialize(@NotNull final IFactoryController controller, @No
     {
         final ItemStack stack = ItemStack.read(nbt.getCompound(TAG_STACK));
         final int size = nbt.getInt(TAG_SIZE);
-        return this.getNewInstance(stack, size);
+        final boolean ignoreNBT = nbt.contains(TAG_SHOULDIGNORENBT) ? nbt.getBoolean(TAG_SHOULDIGNORENBT) : false;
+        final boolean ignoreDamage = nbt.contains(TAG_SHOULDIGNOREDAMAGE) ? nbt.getBoolean(TAG_SHOULDIGNOREDAMAGE) : false;
```

Comment:
`getBoolean` returns false when the tag is missing, so the `contains` checks are not needed.

Reply:
Now, this one I'd forgotten, and just went with the safe pattern. Anyway, it's cleaned up.